_posts/2023-05-09-introducing-timelines-part-1.md

```diff
@@ -0,0 +1,169 @@
+image: assets/images/blog/introducing-timelines-part-1.png
+author: Ben Chambers and Therapon Skoteiniotis
+featured: true
+---
+
+In the last decade, advances in data analytics, machine learning, and AI have occurred, thanks in part to the development of technologies like Apache Spark, Apache Flink, and KSQL for processing streaming data.
+Unfortunately, streaming queries remain difficult to create, comprehend, and maintain -- you must either use complex low-level APIs or work around the limitations of query languages like SQL that were designed to solve a very different problem.
+To tackle these challenges, we introduce a new abstraction for time-based data, called _timelines_.
+
+Timelines organize data by time and entity, offering an ideal structure for event-based data with an intuitive, graphical mental model.
+Timelines simplify reasoning about time by aligning your mental model with the problem domain, allowing you to focus on _what_ to compute, rather than _how_ to express it.
+
+In this post, we present timelines as a natural development in the progression of stream processing abstractions.
+We delve into the timeline concept, its interaction with external data stores (inputs and outputs), and the lifecycle of queries utilizing timelines.
+
+This post is the first in a series introducing Kaskada, an open-source event processing engine designed around the timeline abstraction.
+The full series will include (1) this introduction to the timeline abstraction, (2) how Kaskada builds an expressive temporal query language on the timeline abstraction, (3) how the timeline data model enables Kaskada to efficiently execute temporal queries, and (4) how timelines allows Kaskada to execute any query to incrementally over event streams.
+
+# Rising Abstractions in Stream Processing
+
+Since 2011, alongside the rise of distributed streaming, there has been a trend towards higher abstractions for operating on events.
+In the same year that [Apache Kafka][kafka] introduced its distributed stream, [Apache Storm][storm] emerged with a structured graph representation for stream consumers, enabling the connection of multiple processing steps.
+Later in 2013, [Apache Samza][samza] provided fault-tolerant, stateful operations over streams, and the [MillWheel paper][millwheel] introduced the concept of watermarks to handle late data.
+Between 2016 and 2017, [Apache Flink][flink], [Apache Spark][spark], and [KSQL][ksql] implemented the ability to execute declarative queries—SQL—over streams.
+
+![History of Streaming Abstractions, showing Apache Kafka, Apache Storm, Apache Samza, MillWheel, Apache Flink, Apache Spark and KSQL][stream_abstraction_history]
+
+Each of these developments enhanced the abstractions used when working with streams -- improving aspects such as composition, state, and late-data handling.
+The shift towards higher-level, declarative queries significantly simplifies stream operations and the management of compute systems.
```

Comment:
Change from past-tense to present tense in
* enhanced
* simplifies

It can still work, but I think it should be _simplified,_ to emphasize these are past changes. Then the next sentence works well by saying, "still poses" (presently).